databroker/utils.py

```diff
@@ -414,3 +414,14 @@ def __iter__(self):
     def __contains__(self, k):
         # make sure checking 'in' does not trigger evaluation
         return k in self.__mapping
+
+    def add(self, *args, **kwargs):
+        dictionary = dict(*args, **kwargs)
+        wrap = self.__Wrapper
+        with self.__lock:
+            intersection = set(dictionary.keys()).intersection(self.__mapping.keys())
```

Comment:
Iterating over a dict yields its keys, so this may be tightened to save two method calls.
```suggestion
            intersection = set(dictionary).intersection(self.__mapping)
```

Reply:
updated